src/test/java/htsjdk/samtools/cram/compression/rans/RansTest.java

```diff
@@ -49,107 +57,145 @@ public String toString() {
+        };
     }
 
-    @Test
-    public void testSizeRangeTiny() {
-        for (int i = 0; i < 20; i++) {
-            final byte[] data = randomBytesFromGeometricDistribution(100, 0.1);
-            final ByteBuffer in = ByteBuffer.wrap(data);
-            for (int size = 1; size < data.length; size++) {
-                in.position(0);
-                in.limit(size);
-                roundTripForEachOrder(in);
-            }
-        }
+    @DataProvider(name="rans4x8")
+    public Object[][] getRans4x8Codecs() {
+        final RANS4x8Encode rans4x8Encode = new RANS4x8Encode();
+        final RANS4x8Decode rans4x8Decode = new RANS4x8Decode();
+        return new Object[][]{
+                {rans4x8Encode, rans4x8Decode, new RANS4x8Params(RANSParams.ORDER.ZERO)}, // RANS4x8 Order 0
+                {rans4x8Encode, rans4x8Decode, new RANS4x8Params(RANSParams.ORDER.ONE)} // RANS4x8 Order 1
+        };
+    }
+
+    @DataProvider(name="ransNx16")
+    public Object[][] getRansNx16Codecs() {
+        final RANSNx16Encode ransNx16Encode = new RANSNx16Encode();
+        final RANSNx16Decode ransNx16Decode = new RANSNx16Decode();
+        // TODO: More formatFlags values i.e, combinations of bit flags will be added later
+        return new Object[][]{
+                {ransNx16Encode, ransNx16Decode, new RANSNx16Params(0)} //RANSNx16 formatFlags(first byte) 0
+        };
+    }
+
+    public Object[][] getAllRansCodecs() {
+        // concatenate RANS4x8 and RANSNx16 codecs
+        return Stream.concat(Arrays.stream(getRans4x8Codecs()), Arrays.stream(getRansNx16Codecs()))
+                .toArray(Object[][]::new);
     }
 
-    @Test
-    public void testSizeRangeSmall() {
-        final byte[] data = randomBytesFromGeometricDistribution(1000, 0.01);
-        final ByteBuffer in = ByteBuffer.wrap(data);
-        for (int size = 4; size < data.length; size++) {
+    @DataProvider(name="allRansAndData")
+    public Object[][] getAllRansAndData() {
+        // this data provider provides all the testdata for all of RANS codecs
+        return TestNGUtils.cartesianProduct(getRansTestData(), getAllRansCodecs());
+    }
+
+    @DataProvider(name="allRansAndDataForTinySmallLarge")
+    public Object[][] getAllRansAndDataForTinySmallLarge() {
+        // this data provider provides Tiny, Small and Large testdata for all of RANS codecs
+        return TestNGUtils.cartesianProduct(getRansTestDataTinySmallLarge(), getAllRansCodecs());
+    }
+
+    @Test(dataProvider = "allRansAndDataForTinySmallLarge")
+    public void testSizeRangeTinySmallLarge(
+            final TestCaseWrapper tc,
+            final Integer lowerLimit,
+            final Integer upperLimit,
+            final RANSEncode ransEncode,
+            final RANSDecode ransDecode,
+            final RANSParams params){
+        final ByteBuffer in = ByteBuffer.wrap(tc.testArray);
+        for (int size = lowerLimit; size < upperLimit; size++) {
```

Comment:
I know you didn't add this, but do you understand what the point of these limits are ? It seems redundant to repeatedly do the test over different size fragments of the buffer. Can you see any value to that ?